src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -42,13 +44,19 @@ internal ItemManager(RecentItemsManager recentItemsManager)
             IAsyncCompletionSession session,
             AsyncCompletionSessionInitialDataSnapshot data,
             CancellationToken cancellationToken)
-            => Task.FromResult(data.InitialList.OrderBy(i => i.SortText).ToImmutableArray());
+        {
+            // The same ItemManager can be used multiple times in different completion sessions on the same TextView.
```

Comment:
it's ok to persist this in the session's property bag until I make API change where I provide this piece of data to the Update method.
The issue that needs to be addressed is unnecessary dependency of item manager on item source.

Reply:
@AmadeusW If I understand you correctly, we can take the trigger kind from `Sort`, store it in the session, and retrieve it in `Update` (basically what the field is doing but without having more "state" (kinda) in the ItemManager). Then if the session is reused it'll be fine because we'll get another `Sort` before the next `Update`. Is this right?

I know Sort will always be called before Update. Are there any threading concerns here though?